cmd/zpool/zpool_iter.c

```diff
@@ -330,10 +407,13 @@ vdev_run_cmd_thread(void *cb_cmd_data)
 	FILE *fp;
 	size_t len = 0;
 	char cmd[_POSIX_ARG_MAX];
+	char *line;
 
 	/* Set our VDEV_PATH and VDEV_UPATH env vars and run command */
-	if (snprintf(cmd, sizeof (cmd), "VDEV_PATH=%s && VDEV_UPATH=\"%s\" && "
-	    "VDEV_ENC_SYSFS_PATH=\"%s\" && %s", data->path ? data->path : "",
+	if (snprintf(cmd, sizeof (cmd), "export VDEV_PATH=%s && "
+	    "export VDEV_UPATH=\"%s\" && "
+	    "export VDEV_ENC_SYSFS_PATH=\"%s\" && %s",
+	    data->path ? data->path : "",
```

Comment:
This looks like it can be simplified too.

```
        /* Set our VDEV_* environment variables and run the command */
        len = snprintf(cmd, sizeof (cmd), 
            "VDEV_PATH=\"%s\" VDEV_UPATH=\"%s\" VDEV_ENC_SYSFS_PATH=\"%s\" %s",
            data->path ? data->path : "",
            data->upath ? data->upath : "",
            data->vdev_enc_sysfs_path ? data->vdev_enc_sysfs_path : "",
            data->cmd);
        if (len >= sizeof (cmd)) {
        }
```